src/AWS.Deploy.Orchestration/Data/AWSResourceQueryer.cs

```diff
@@ -167,6 +167,46 @@ await foreach (var response in listInstanceTypesPaginator.Responses)
             });
         }
 
+        public async Task<InstanceTypeInfo?> DescribeInstanceType(string instanceType)
+        {
+            var ec2Client = _awsClientFactory.GetAWSClient<IAmazonEC2>();
+
+            var request = new DescribeInstanceTypesRequest
+            {
+                InstanceTypes = new List<string> { instanceType }
+            };
+
+            return await HandleException(async () =>
+            {
+                try
+                {
+                    var response = await ec2Client.DescribeInstanceTypesAsync(request);
+
+                    if (response.InstanceTypes?.Count >= 1)
+                    {
+                        return response.InstanceTypes[0];
+                    }
+                    else
+                    {
+                        return null;
+                    }
```

Comment:
nit: could you do this using Linq? (return response.InstanceTypes.FirstOrDefault();)

Reply:
Done